cpp/oneapi/dal/backend/common.hpp

```diff
@@ -160,11 +160,37 @@ inline void check_if_same_context(const sycl::queue& q1,
 /// Creates `nd_range`, where global size is multiple of local size
 inline sycl::nd_range<1> make_multiple_nd_range_1d(std::int64_t global_size,
                                                    std::int64_t local_size) {
-    const std::int64_t g = dal::detail::integral_cast<std::size_t>(global_size);
-    const std::int64_t l = dal::detail::integral_cast<std::size_t>(local_size);
+    const std::size_t g = dal::detail::integral_cast<std::size_t>(global_size);
+    const std::size_t l = dal::detail::integral_cast<std::size_t>(local_size);
     return { up_multiple(g, l), l };
 }
 
+template <std::int64_t axis_count>
+using ndindex = std::array<std::int64_t, axis_count>;
```

Comment:
maybe nd_index for consistency?
What about 1-dimensional index?

Reply:
I took the ndindex using from ndarray, for consistency )), it is also consistent with ndarray, seems we need to finalize the approach for naming.

Reply:
i wouldn't introduce extra entity in 1d case just for having common look

Reply:
need to have an agreement regarding naming and change it as separate MR